drivers/SmartThings/philips-hue/src/utils.lua

```diff
@@ -64,10 +64,102 @@ function utils.backoff_builder(max, inc, rand)
   end
 end
 
+function utils.labeled_socket_builder(label)
```

Comment:
Where was all this happening before?  What was this change meant to handle?

Reply:
This is a direct copy over from the default socket builder in `rest.lua` (the function call `make_socket`).

@kianooshkarami had mused about being able to have more context in some of the low level rest calls so that we can know e.g. which bridge we're having failures on.

The socket builder itself isn't new functionality, all the REST stuff already uses it, we were just inlining the default one in the rest.lua module and now instead we create the function via this builder so that we can prefix it with a string label and get more context.

https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/799/files#diff-7e9e351ebde6cef0a6366cf2bc69bd47ec79864fdf869ba8c8964502046c9f45L237